### ISSUE TITLE
Possibility to point directly to a chart on github or similar

### ISSUE THREAD

**Issue by timdeluxe** [hashicorp/terraform-provider-helm#465]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

i want the option to point directly to a chart on github or similar.

Example: I want to use https://github.com/dmitrii-ageev/helm-teamcity, which is not part of any helm repository

I tried every way that came into my mind: 
- configuring that target repo as "chart"
- configuring that repo as "repository"
- Pointing to the released tar.gz of the chart directly in both of them
- Using a separate helm_repository data resource

...none of them worked (mostly the download failed or it expected a non-existing index.yaml).

I ended up in downloading the chart and putting it into my own repo near to the terraform code and pointing to it with a path.

### Potential Terraform Configuration

```hcl
resource "helm_release" "teamcity" {
  name = "teamcity"
  chart = "https://github.com/dmitrii-ageev/helm-teamcity/archive/1.0.7.tar.gz"
...
```

### References
n/a

**Comment by jrhouston** [comment]
I think this is actually a bug, because the path formats that the `helm install` command supports should also be supported by this provider.

**Comment by jrhouston** [comment]
Oh interesting, so your example does actually work for me - although the install fails because the chart is using APIs older than my local cluster supports. Which version of the provider are you using?

**Comment by timdeluxe** [comment]
@jrhouston 
In the case for which i opened the issue for it was indeed 1.0.0, however i tried to reproduce it with the K8S cluster bundled with docker desktop for mac and there i tried it also with provider version 1.1.1 and still get an error:
```
Error: failed to download "https://github.com/dmitrii-ageev/helm-teamcity/archive/1.0.7.tar.gz" (hint: running `helm repo update` may help)
```
With which provider version does it work for you?

**Comment by woz5999** [comment]
I get this with both 1.0.0 and 1.1.1

**Comment by woz5999** [comment]
After setting debug=true in the helm provider, I was able to get some meaningful error messages from the provider:

```
Error: no cached repo found. (try 'helm repo update'): open /Users/[username]/Library/Caches/helm/repository/stable-index.yaml: no such file or directory
```

Running ```helm repo update``` populated the stable index and then I started to receive the same error as above except now for "local-index.yaml". Creating local-index.yaml at the path referenced in the error message and with the below contents resolved the issue for me:

```
apiVersion: v1
entries:
```

The logic that throws this error is deep in the vendored helm client code, but given that the provider/helm_release resource allow you to specify the repository inline and there's no documentation specifying anything about these local repo caches, this is a pretty miserable and difficult to troubleshoot bug. I'm not sure if the answer is an improvement to the provider or just better documentation, but this is going to cause people a lot of pain, especially now that the repository data source is deprecated.
